packages/nc-gui-v2/components/dashboard/settings/UIAcl.vue

```diff
@@ -0,0 +1,146 @@
+}
+
+onMounted(async () => {
+  if (tables.length === 0) {
+    await loadTableList()
+  }
+})
+
+const tableHeaderRenderer = (label: string) => () => h('div', { class: 'text-gray-500' }, label)
+
+const columns = [
+  {
+    title: tableHeaderRenderer('Table name'),
+    name: 'table_name',
+  },
+  {
+    title: tableHeaderRenderer('View name'),
+    name: 'view_name',
+  },
+  {
+    // { icon: MdiGridIcon, color: 'blue' },
+    title: tableHeaderRenderer('Editor'),
+    name: 'editor',
+    width: 150,
+  },
+  {
+    title: tableHeaderRenderer('Commenter'),
+    name: 'commenter',
+    width: 150,
+  },
+  {
+    title: tableHeaderRenderer('Viewer'),
+    name: 'viewer',
+    width: 150,
+  },
+]
+</script>
+
+<template>
+  <div class="flex flex-row w-full">
+    <div class="flex flex-column w-full">
+      <div class="flex flex-row items-center w-full mb-4 gap-2">
+        <a-input v-model:value="searchInput" placeholder="Search models">
+          <template #prefix>
+            <MdiMagnify />
+          </template>
+        </a-input>
+        <a-button class="self-start" @click="loadTableList">
+          <div class="flex items-center gap-2 text-gray-600 font-light">
+            <MdiReload :class="{ 'animate-infinite animate-spin !text-success': isLoading }" />
+            Reload
+          </div>
+        </a-button>
+        <a-button class="self-start" @click="saveUIAcl">
+          <div class="flex items-center gap-2 text-gray-600 font-light">
+            <MdiContentSave />
+            Save
+          </div>
+        </a-button>
+      </div>
+      <a-table
+        class="w-full"
+        :data-source="
+          tables.filter(
+            (el) =>
+              (typeof el?._ptn === 'string' && el._ptn.toLowerCase().includes(searchInput.toLowerCase())) ||
+              (typeof el?.title === 'string' && el.title.toLowerCase().includes(searchInput.toLowerCase())),
+          )
+        "
+        :columns="columns"
+        :pagination="false"
+        :loading="isLoading"
+        bordered
+      >
+        <template #bodyCell="{ record, column }">
+          <div v-if="column.name === 'table_name'">{{ record._ptn }}</div>
+          <div v-if="column.name === 'view_name'">
+            <div class="flex align-center">
+              <component :is="viewIcons[record.type].icon" :class="`text-${viewIcons[record.type].color} mr-1`" />
+              {{ record.title }}
+            </div>
+          </div>
+          <div v-for="role in roles" :key="role">
+            <div v-if="column.name === role">
+              <a-tooltip>
+                <template #title>Click to hide '{{ record.title }}' for role:{{ role }} in UI dashboard</template>
+                <a-checkbox
+                  :checked="!record.disabled[role]"
+                  @change="
+                    // eslint-disable-next-line prettier/prettier
+                    record.disabled[role] = !record.disabled[role];
+                    record.edited = true
```

Comment:
Don't do this.
Events are meant to receive a function not just some JS code.

Add a Event handler like 
```ts
const onChange = () => {
 	record.disabled[role] = !record.disabled[role];
	record.edited = true
}
```

```vue
<a-checkbox  @change="onChange" />
```